### DISCUSSION TITLE
Drink Water App

### DISCUSSION THREAD

**Discussion by Cotrox** [zepp-health/zeppos-forum#100] [User Feedback]
Hi guys! I really love Drink Water App but I have to tell you some problems.

1. Notification Problem: Notifications are activate but usually they never arrive to me, it works just sometimes.
2. If you set a different ml value (for exemple 3000 instead of 2k) adding ml over 2k (like 2250/3000) never save the variation. When you come back to the app you will see 2k/3k.

**Comment by MyoungXUE** [comment]
Thank you for your feedback, I will feedback this issue to the relevant colleagues.

**Reply by JohnnyGo-x** [reply]
Hi, Is drink app using “setTimeout” api to send notification at the specific time?

**Reply by JohnnyGo-x** [reply]
I have a similar demand to send notification in specific time even when app is not running in the foreground.
I notice ZeppOS 3.0 support backgroud tasks but the former version not. Wonder how "Water Drink" app did that. Thanks

**Reply by Xuemuyang** [reply]
@JohnnyGo-chn 
Hello, the drinking Mini Program uses the `hmApp.alarmNew` API, which can wake up a page of the Mini Program at regular intervals.

**Comment by Xuemuyang** [comment]
@Cotrox 
Hi, in the current version of Zepp OS, some of the timers used by the applets are not persistent, which may cause timer loss if the watch is restarted. This issue is being optimized in Zepp OS 3.0.
